source/blog/2014-02-04-ember-1-10-0-released.md

```diff
@@ -0,0 +1,387 @@
+implementing this feature.
+
+#### Bound Attribute Syntax
+
+Current Ember developers are familiar with the `bind-attr` syntax, used
+to declare an attribute binding on an HTML element. An original
+motivation for HTMLBars was to improve on this syntax.
+
+Ember 1.11 introduces a more intuitive API for attribute binding. For
+example, here the `color` variable is bound to the class of a div:
+
+```hbs
+<div class="{{color}}"></div>
+```
+
+The inline if helper can also be used in these contexts:
+
+```hbs
+<div class="{{color}} {{if isEnabled 'active' 'disabled'}}"></div>
+```
+
+For some attributes, like the `disabled` boolean, passing a literal value
+is desirable. An example:
+
+```hbs
+<input disabled={{isDisabled}}>
+```
+
+To allow the data-binding of non-string values to boolean properties and
+custom element properties, bound attributes are implemented with a
+**property-first** setter.
+
+When binding to an attribute, Ember first checks to see if that attribute is a
+property of the element's DOM node (with normalization of capitalization). If it
+is, the value is set with a property. For example:
+
+```hbs
+<input disabled={{isDisabled}}>
+```
+
+```js
+// disabled is a property of input elements, so...
+input.disabled = true;
+```
+
+If the attribute is not present as a property, then its value is set as an
+attribute:
+
+```hbs
+<div class={{color}}>
+```
+
+```js
+// class is not a property of div elements, do...
+div.setAttribute('class', 'red');
+```
+
+For SVG attributes and the style attribute, we've made a exceptions to this pattern.
+Despite these exceptions, the property-first rule is a good way to describe what is
+happening behind the scenes. In practice, the binding syntax feels natural.
+
+Many thanks to [@mixonic](http://twitter.com/mixonic), [@\_mmun](http://twitter.com/_mmun),
+and [@wycats](http://twitter.com/wycats) for their effort on the design and implementation
+of this feature.
+
+#### Named Substates
+
+Two routing substates exist for Ember routes. The `loading` substate will be entered
+if the async hooks of a route are still processing, and the `error` substate will be
+entered when an async hook promise is rejected.
+
+Substates are sibling routes. When the `new` route of `cars.new` is loading, the `cars.loading`
+substate is entered. The `application` route has no parent namespace to nest its siblings.
+This makes using the `loading` and `error` substates impossible for an application route.
+
+Name substates add a new lookup method for substates. The name of the route is pre-pended
+onto the substate. So a valid loading substate for `application` can be defined as
+`application_loading`.
+
+Thanks to [@machty](http://twitter.com/machty) for landing this feature.
+
+#### Component Helper
+
+Ember components can be bound via the `component` helper. For example this logic
+in a template:
+
+```hbs
+{{#if isRed}}
+  {{x-red}}
+{{else if isBlue}}
```

Comment:
This is not valid syntax in 1.0 through 1.9 (it was enabled by HTMLBars work).  Doesn't really matter, but I figured I'd point it out...